lib/util.ml

```diff
@@ -224,6 +224,36 @@ let search_in_path p s =
 
 let search_in_assets = search_in_path Secure.assets
 
+let hash_file file =
+  if Sys.file_exists file then
+    try
+      let ic = open_in_bin file in
+      Fun.protect
+        ~finally:(fun () -> close_in ic)
```

Comment:
As usual, you shouldn't call `close_in` in a finalizer. Use `Compat.In_channel.with_open_bin` instead ;)